File: native/mixdog-graph/tests/fixtures/outline/zig/sample.zig
const std = @import("std");
const util = @import("./util.zig");

pub const Config = struct {
    name: []const u8,
    enabled: bool,

    pub fn init(name: []const u8) Config {
        return .{ .name = name, .enabled = true };
    }
};

const Mode = enum { fast, slow };

const Payload = union(enum) {
    none,
    bytes: []const u8,
};

pub fn greet(name: []const u8) []const u8 {
    _ = name;
    return "hi";
}

fn helper(n: usize) usize {
    return n + 1;
}

pub fn main() void {
    const cfg = Config.init("demo");
    _ = helper(cfg.name.len);
    _ = greet("world");
    _ = Mode.fast;
    _ = Payload.none;
}

File: native/mixdog-graph/tests/fixtures/outline/r/sample.r
library(ggplot2)
library("dplyr")
require(stats)
source("helpers.R")
source('util/extra.R')

greet <- function(name) {
  paste("hello", name)
}

add = function(a, b) {
  a + b
}

plot_counts <- function(df) {
  ggplot(df, aes(x = name, y = n)) + geom_col()
}

helper <- function(x) {
  x + 1
}

main <- function() {
  greet("world")
  helper(add(1, 2))
}

not_a_function <- 42

# Ordinary value assignments are not function symbols.
scale_factor <- 1.5
title <- "demo plot"

# Top-level helper. A `<- function` assignment nested inside `main` would be
# a CLI item-pruning loss (see rules/outline/PARITY-NOTES.md).
inner_add <- function(x, y) {
  x + y
}

